lib/src/bluez_client.dart

```diff
@@ -340,6 +344,239 @@ class BlueZAdapter {
           .toList();
 }
 
+/// BtLE Advertising Manager
+class BlueZAdvertisingManager {
+  final String _advertInterfaceName = 'org.bluez.LEAdvertisingManager1';
+
+  final BlueZClient _client;
+  final _BlueZObject _object;
+
+  BlueZAdvertisingManager(this._client, this._object);
+
+  Future<void> registerAdvertisement(BlueZAdvertisement advert) async {
+    await _client._bus.registerObject(advert);
+
+    await _object.callMethod(_advertInterfaceName, 'RegisterAdvertisement',
+        [advert.path, DBusDict.stringVariant({})],
+        replySignature: DBusSignature(''));
+  }
+
+  Future<void> unregisterAdvertisement(BlueZAdvertisement advert) async {
+    await _object.callMethod(
+        _advertInterfaceName, 'UnregisterAdvertisement', [advert.path],
+        replySignature: DBusSignature(''));
+
+    await _client._bus.unregisterObject(advert);
+  }
+}
+
+/// BtLE Advertisement instance
+abstract class BlueZAdvertisement extends DBusObject {
+  final String _advertInterfaceName = 'org.bluez.LEAdvertisement1';
+
+  BlueZAdvertisement(DBusObjectPath path) : super(path);
+
+  Future<void> release();
+
+  Map<BlueZManufacturerId, DBusValue> get manufacturerData;
+  String get type;
```

Comment:
Are the types fixed? If so change to an enum.

Reply:
Yes, I'll make it an enum.

Reply:
Done